src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodo, setCurrentTodo] = useState<Todo | undefined>();
+  const [currentUser, setCurrentUser] = useState<User | undefined>();
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    let currentTodos = todos;
+    const formatedQuery = query.trim().toLowerCase();
+
+    if (filterType !== 'all') {
+      currentTodos = currentTodos.filter((todo) => {
+        const { completed } = todo;
+
+        if ((filterType === 'completed' && completed)
+          || (filterType === 'active' && !completed)) {
+          return true;
+        }
+
+        return false;
+      });
+    }
+
+    if (formatedQuery) {
+      currentTodos = currentTodos.filter((todo) => (todo.title.toLowerCase().includes(formatedQuery)));
+    }
+
+    return currentTodos;
+  }, [todos, query, filterType]);
+
+  const handleSelectTodo = (todo: Todo) => {
+    setCurrentTodo(todo);
+    getUser(todo.userId)
+      .then(setCurrentUser);
+  };
+
+  const handleCloseModal = useCallback(() => {
+    setCurrentTodo(undefined);
+    setCurrentUser(undefined);
```

Comment:
You should always set it to `null` instead of `undefined` if you set it manually

Reply:
I used undefined to take care of errors saying they cannot assign "Todo | null | undefined" to "Todo | null", it was the simpler solution; I will attempt to mitigate

Reply:
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/55781950/0c63f65c-74fd-48ad-96e2-d900c3266b84)
I mean this error right here